proposals/lip-0045.md

```diff
@@ -779,154 +761,41 @@ The `apply` function applies a CCM. It is specified in [LIP 0049][lip-0049].
-    ccm = {
-        "nonce": ownChainAccount.nonce,
-        "moduleID": moduleID,
-        "crossChainCommandID": crossChainCommandID,
-        "sendingChainID": ownChainAccount.ID,
-        "receivingChainID": receivingChainID,
-        "fee": fee,
-        "status": status,
-        "params": params
-    }
-
-    if validateFormat(ccm) == False:
-        return False
-
-    for each module mdl for which beforeSendCCM exists:
-        mdl.beforeSendCCM(feeAddress, ccm)
-        if the previous call fails:
-            return False
-
-    addToOutbox(partnerChainID, ccm)
-    ownChainAccount.nonce += 1
-
-    return True
-```
-
-#### terminateChainInternal
-
-The `terminateChainInternal` function terminates a chain account.
-
-##### Parameters
-
-* `chainID`: The ID of the chain to be terminated.
-
-##### Returns
-
-A boolean indicating the success of the function execution.
-
-##### Execution
-
-```python
-terminateChainInternal(chainID):
-    sendInternal(
-        EMPTY_FEE_ADDRESS,
-        MODULE_ID_INTEROPERABILITY,
-        CROSS_CHAIN_COMMAND_ID_CHANNEL_TERMINATED,
-        chainID,
-        0,
-        CCM_STATUS_OK,
-        EMPTY_BYTES
-    )
-
-    return createTerminatedStateAccount(chainID)
-```
-
 #### createTerminatedStateAccount
 
 The `createTerminatedStateAccount` function creates an entry in the terminated state substore.
 
-##### Parameters
-
-* `chainID`: The ID of the chain.
-* `stateRoot`: The state root to be stored. If none is passed, it is deduced from the state.
-
-##### Returns
-
-A boolean indicating the success of the function execution.
-
-##### Execution
-
 ```python
-createTerminatedStateAccount(chainID, stateRoot = None):
-    if stateRoot:
-        if chainAccount(chainID) exists:
-            chainAccount(chainID).status = CHAIN_TERMINATED
-            remove the entry with storeKey = uint32be(chainID) from the outbox root substore
+def createTerminatedStateAccount(chainID: ChainID, stateRoot: bytes = EMPTY_BYTES) -> None:    
+    if chainAccount(chainID) exists:
+        chainAccount(chainID).status = CHAIN_STATUS_TERMINATED
+        remove the entry with storeKey = chainID from the outbox root substore
+
+        # If no stateRoot is given as input, get it from the state
+        if len(stateRoot) == 0:
+            stateRoot = chainAccount(chainID).lastCertificate.stateRoot
 
         terminatedState = {
             "stateRoot": stateRoot,
             "mainchainStateRoot": EMPTY_BYTES,
             "initialized": True
         }
 
-    elif chainAccount(chainID) exists:
-        chainAccount(chainID).status = CHAIN_TERMINATED
-        remove the entry with storeKey = uint32be(chainID) from the outbox root substore
-
-        terminatedState = {
-            "stateRoot": chainAccount(chainID).lastCertificate.stateRoot,
-            "mainchainStateRoot": EMPTY_BYTES,
-            "initialized": True
-        }
+        # Emit chain status updated event.
+        emitEvent(
```

Comment:
I am not sure it is so nice here to only have an event in this case, but not the case below.

Reply:
The idea is that we emit the event whenever we update the chain account. On a sidechain there is no chain account to update tho. One option would be to introduce a new event when the terminated state account is created

Reply:
I think it would be useful to have an event here. Otherwise, it may be difficult to find out at which stage and for what reason a chain is terminated.

Reply:
added a new event for creating chain terminated account

Reply:
Looks good, I resolve this comment.